Shell dummy backend (#10660)
zephyrproject-rtos/zephyr: doc/subsystems/shell/shell.rst

At this point, the following transport layers are implemented:

* UART

> * DUMMY - not phisical transport layer

Reviewer: ```suggestion
* DUMMY - not a physical transport layer
```

> * DUMMY - not a phisical transport layer

Reviewer: change ``phisical`` to ``physical``

> * DUMMY - not a phisical transport layer

Reviewer: change ``phisical`` to ``physical``

handler) are passed as arguments. Characters within parentheses are treated
as one argument. If shell wont find a handler it will display an error message.

> Commands can be also executed from user's application using DUMMY backend and

Reviewer: ```suggestion
Commands can be also executed from a user application using the DUMMY backend and
```

> a function :cpp:func:`shell_execute_cmd` like presented in below example:

Reviewer: ```suggestion
a function :cpp:func:`shell_execute_cmd`, as shown in this example:
```

.. code-block:: c

	void main(void)
	{
		/* Below code will execute "root_cmd subcommand" command */

> 		shell_execute_cmd(NULL, "root_cmd subcommand param");

Reviewer: This would be a better example if you showed a specific command and its parameters.

	}

> DUMMY backend needs to be enabled in Kconfig in order to use it.

Reviewer: ```suggestion
Enable the DUMMY backend by setting the Kconfig :option:`CONFIG_SHELL_BACKEND_DUMMY` option.
```
